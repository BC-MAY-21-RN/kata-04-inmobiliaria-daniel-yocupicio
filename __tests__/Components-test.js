import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../src/components/CardComponent';
import List from '../src/components/ListComponent';

describe('Components test', () => {
  it('Card component snapshot', () => {
    const renderCard = renderer
      .create(
        <Card
          item={{
            name: 'The Game',
            adress: '3517 The Best Game',
            rooms: 4,
            bathrooms: 2,
            size: 230,
            cost: 200,
            evaluation: 4.9,
            like: false,
          }}
          islast={false}
        />,
      )
      .toJSON();
    expect(renderCard).toMatchSnapshot();
  });
  it('List component snapshot', () => {
    const renderList = renderer.create(<List data={[]} />).toJSON();
    expect(renderList).toMatchSnapshot();
  });
});
