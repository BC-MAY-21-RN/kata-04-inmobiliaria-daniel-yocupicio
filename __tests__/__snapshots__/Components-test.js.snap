// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components test Card component snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5fdff",
      "borderRadius": 20.05,
      "height": 140,
      "marginTop": 15,
      "width": 710,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingHorizontal": 10,
      }
    }
  >
    <View>
      <
        height={120}
        width={120}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#fbedb7",
            "borderRadius": 24,
            "bottom": 5.7,
            "flex": 1,
            "flexDirection": "row",
            "paddingHorizontal": 11,
            "paddingVertical": 4,
            "position": "absolute",
          }
        }
      >
        <
          height={15}
          width={15}
        />
        <Text
          style={
            Object {
              "color": "#7a6229",
              "marginLeft": 3,
            }
          }
        >
          4.9
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 120,
          "marginLeft": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#151525",
            "fontSize": 19,
          }
        }
      >
        The Game
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 21,
          }
        }
      >
        <
          height={19}
          width={19}
        />
        <Text
          style={
            Object {
              "fontSize": 13,
              "marginLeft": 5,
              "maxWidth": 130,
            }
          }
        >
          3517 The Best Game
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <
          height={24}
          width={24}
        />
        <Text
          style={
            Object {
              "color": "#151525",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginLeft": 4,
              "marginRight": 6,
            }
          }
        >
          4
        </Text>
        <
          height={24}
          width={24}
        />
        <Text
          style={
            Object {
              "color": "#151525",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginLeft": 4,
              "marginRight": 6,
            }
          }
        >
          2
        </Text>
        <
          height={24}
          width={24}
        />
        <Text
          style={
            Object {
              "color": "#151525",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginLeft": 4,
            }
          }
        >
          230
           ft
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        $ 
        200
        /m
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#00B074",
        "borderRadius": 50,
        "bottom": 6,
        "padding": 7,
        "position": "absolute",
        "right": 10,
      }
    }
  >
    <
      height={16}
      width={16}
    />
  </View>
</View>
`;

exports[`Components test List component snapshot 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;
